code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -57,9 +57,12 @@
 	QDEL_NULL(internal_gps)
 	return ..()
 
+/mob/living/simple_animal/hostile/megafauna/Life(seconds, times_fired)
+	. = ..()
+	do_chasm()
```

Comment:
on life AND when moving? bruh

Reply:
I agree, this is a fuck-ton of timer calls

Reply:
any ideas for a better suggestion to avoid the timer call issue, that will add up fast if surrounded by chasms yeah

Reply:
Yeah, refactor chasms to not be dense, and hostile mob pathfinding to avoid them

Reply:
This can be done relatively easily:
Insert an exception and redirect to certain turf types in `/mob/living/simple_animal/proc/handle_automated_movement()`, add megafauna alongside `if(M.flying || M.floating)` within `/turf/simulated/floor/chasm/proc/droppable(atom/movable/AM)`